src/test/scala/mesosphere/marathon/integration/setup/MesosTest.scala

```diff
@@ -57,9 +81,40 @@ case class MesosCluster(
     // to it's marathon, leading to multiple tasks (from different IT suits) trying to use the same port!
     // First-come-first-served task will bind successfully where the others will fail leading to a lot inconsistency and
     // flakiness in tests.
+
+    val (faultDomainAttributes: Map[String, Option[String]], faultDomainJson) = if (agentsFaultDomains.nonEmpty && agentsFaultDomains(i).nonEmpty) {
+      val faultDomain = agentsFaultDomains(i).get
+      val faultDomainJson = s"""
```

Comment:
Hm, I have a hard time understand the type of `faultDomainJson`. 

What is

```scala
         |}
        """.stripMargin
      (
        Map(
        "fault_domain_region" -> Some(faultDomain.region.value),
        "fault_domain_zone" -> Some(faultDomain.zone.value)),
        Some(faultDomainJson)
      )
```

doing?

Reply:
It provides agent attributes and cmd option for enabling fault domain in mesos